### PR TITLE
feat(environment-variables): Expand env vars from system env, not only .env file

### PR DESCRIPTION
**Pull request by janpio**
Current description did not make this explicit, making replies like https://github.com/prisma/prisma/issues/3724#issuecomment-905383273 harder.

### CODE REVIEW

**Review by molebox — APPROVED** [review]
miyagi

**Comment by molebox** [comment]
I think something has gone wrong here, maybe you are missing some closing codeblock backticks? 

https://deploy-preview-2208--prisma2-docs.netlify.app/docs/guides/development-environment/environment-variables#expanding-variables

**Review by molebox — APPROVED** [review]
Just need to fix those closing backticks

so-close

**Comment by janpio** [comment]
Hmm meh, the terminal and file view of this does not actually make sense :/

image

**Comment by janpio** [comment]
Better: 
image


Should we add 2 sub headlines to better split the different things apart?

File: content/300-guides/125-development-environment/050-environment-variables/index.mdx [hunk 1]
```diff
@@ -77,14 +77,30 @@ Refer to the `dotenv` documentation for information about [what happens if an en
...
 ### Expanding variables
 
-Variables stored in `.env` files can be expanded using the format specified by [dotenv-expand](https://github.com/motdotla/dotenv-expand). This is useful when dealing with PaaS or similar products. such as Heroku):
+Variables stored in `.env` files can be expanded using the format specified by [dotenv-expand](https://github.com/motdotla/dotenv-expand). 
 
 ```
 # .env
 DATABASE_URL=postgresql://test:test@localhost:5432/test
 DATABASE_URL_WITH_SCHEMA=${DATABASE_URL}?schema=public
 ```
```
Comment (molebox) [inline content/300-guides/125-development-environment/050-environment-variables/index.mdx]:
````suggestion
```env file=.env
DATABASE_URL=postgresql://test:test@localhost:5432/test
DATABASE_URL_WITH_SCHEMA=${DATABASE_URL}?schema=public
```
````

File: content/300-guides/125-development-environment/050-environment-variables/index.mdx [hunk 2]
```diff
@@ -77,14 +77,30 @@ Refer to the `dotenv` documentation for information about [what happens if an en
...
 
+This will make the environment variable `DATABASE_URL_WITH_SCHEMA` with value `postgresql://test:test@localhost:5432/test?schema=public` available for Prisma.
+
+You can also use environment variables in the expansion that are set _outside_ of the `.env` file, for example a database URL that is set on a PaaS like Heroku or similar:
+
+```terminal
+# environment variable already set in the environment of the system
+export DATABASE_URL=postgresql://test:test@localhost:5432/test
+```
```
Comment (molebox) [inline content/300-guides/125-development-environment/050-environment-variables/index.mdx:94]:
````suggestion
```env file=.env
# environment variable already set in the environment of the system
export DATABASE_URL=postgresql://test:test@localhost:5432/test
```
````

File: content/300-guides/125-development-environment/050-environment-variables/index.mdx [hunk 3]
```diff
@@ -77,14 +77,30 @@ Refer to the `dotenv` documentation for information about [what happens if an en
...
+
+```
+# .env
+DATABASE_URL_WITH_SCHEMA=${DATABASE_URL}?schema=foo
+```
```
Comment (molebox) [inline content/300-guides/125-development-environment/050-environment-variables/index.mdx]:
````suggestion
```env file=.env
DATABASE_URL_WITH_SCHEMA=${DATABASE_URL}?schema=foo
```
````

**Review by molebox — APPROVED** [review]
Yep, my bad on terminal. This is the correct way 😅

File: content/300-guides/125-development-environment/050-environment-variables/index.mdx (hunk 1 above)
Comment (janpio) [reply content/300-guides/125-development-environment/050-environment-variables/index.mdx]: Do we have an example in the docs already what that looks like? Would like to see it this time before merging if possible :D

Comment (molebox) [reply content/300-guides/125-development-environment/050-environment-variables/index.mdx]: Sure:
fileenv

File: content/300-guides/125-development-environment/050-environment-variables/index.mdx (hunk 2 above)
Comment (janpio) [reply content/300-guides/125-development-environment/050-environment-variables/index.mdx:94]: No, this is actually a command you execute in the terminal to set the env var

**Pull request merged**